src/Amazon.Common.DotNetCli.Tools/Utilities.cs

```diff
@@ -195,14 +195,78 @@ public static string LookupTargetFrameworkFromProjectFile(string projectLocation
             return element?.Value;
         }
 
-        public static string LookupOutputTypeFromProjectFile(string projectLocation)
+        /// <summary>
+        /// Uses `dotnet msbuild` to determine project properties.
+        /// </summary>
+        /// <param name="projectLocation">Path of the project</param>
+        /// <param name="propertyName">The name of the property</param>
+        /// <returns>The value of the project property</returns>
+        private static string LookupMsbuildProperty(string projectLocation, string propertyName)
         {
-            var projectFile = FindProjectFileInDirectory(projectLocation);
+            var process = new Process();
+            string outputJson;
+            try
+            {
+                process.StartInfo = new ProcessStartInfo()
+                {
+                    FileName = "dotnet",
+                    Arguments = $"msbuild {projectLocation} -getProperty:{propertyName}",
+                    RedirectStandardOutput = true,
+                    UseShellExecute = false,
+                    WindowStyle = ProcessWindowStyle.Hidden
+                };
+
+                process.Start();
+                outputJson = process.StandardOutput.ReadToEnd();
+                var hasExited = process.WaitForExit(5000);
+
+                // If it hasn't completed in the specified timeout, stop the process and give up
+                if (!hasExited) 
+                {
+                    process.Kill();
+                    return null;
+                }
+            }
+            catch (Exception)
+            {
+                // swallow any exceptions related to `dotnet msbuild`
+                return null;
+            }
 
-            var xdoc = XDocument.Load(projectFile);
+            // If it has completed but unsuccessfully, give up
+            if (process.ExitCode != 0)
+            {
+                if (outputJson.Contains("MSBUILD : error MSB1001: Unknown switch."))
```

Comment:
Lets move the fallback logic here and use it anytime the `ExitCode` is non `0`. That will reduce the risk of users seeing any change of behavior. For now I'm fine with just the look up for `OutputType` only using this logic. Once we publish and have and we haven't seen any issues from the change we can switch other msbuild property look ups to this method.

Reply:
Done